policy/nomad/source.go

```diff
@@ -102,41 +104,26 @@ func (s *Source) MonitorIDs(ctx context.Context, req policy.MonitorIDsReq) {
 		// Perform a blocking query on the Nomad API that returns a stub list
 		// of scaling policies. The call is done in a goroutine so we can
 		// still listen for the context closing or a reload request.
-		blockingQueryCompleteCh := make(chan struct{})
-		go func() {
-			// Obtain a handler now so we can release the lock before starting
-			// the blocking query.
-			s.nomadLock.RLock()
-			scaling := s.nomad.Scaling()
-			s.nomadLock.RUnlock()
+		//blockingQueryCompleteCh := make(chan struct{})
 
-			policies, meta, err = scaling.ListPolicies(q)
-			close(blockingQueryCompleteCh)
-		}()
+		// Obtain a handler now so we can release the lock before starting
+		// the blocking query.
+		s.nomadLock.RLock()
+		scaling := s.nomad.Scaling()
+		s.nomadLock.RUnlock()
 
-		select {
-		case <-ctx.Done():
-			s.log.Trace("stopping ID subscription")
-			return
-		case <-s.reloadCh:
-			s.log.Trace("reloading policies")
-			continue
-		case <-blockingQueryCompleteCh:
-		}
+		policies, meta, err = scaling.ListPolicies(q)
+		//close(blockingQueryCompleteCh)
 
+		/*
+		 */
 		// If we get an errors at this point, we should sleep and try again.
 		if err != nil {
-			policy.HandleSourceError(s.Name(), fmt.Errorf("failed to call the Nomad list policies API: %v", err), req.ErrCh)
-			select {
-			case <-ctx.Done():
-				s.log.Trace("stopping ID subscription")
-				return
-			case <-s.reloadCh:
-				s.log.Trace("reloading policies")
-				continue
-			case <-time.After(10 * time.Second):
-				continue
-			}
+			policy.HandleSourceError(s.Name(),
+				fmt.Errorf("failed to call the Nomad list policies API: %v", err), req.ErrCh)
+			time.Sleep(10 * time.Second)
+
```

Comment:
we also don't want to sleep a full 10 seconds, or bother error-handling something that may be stale from a reload, or irrelevant from a shutdown, eh?